### ISSUE TITLE
Confusing constructor for DiskDataset

### ISSUE THREAD

**Issue by peastman** [deepchem/deepchem#321]
I find the DiskDataset constructor rather confusing.  The reason is that a single constructor is used for two quite different functions: loading an existing dataset from disk, and creating a new dataset on disk.  They differ only in the constructor arguments, primarily through the `reload` flag.  To load an existing dataset you write

```Python
dataset = DiskDataset(data_dir='datadir', reload=True)
```

To create a new dataset on disk, you write

```Python
dataset = DiskDataset(data_dir='datadir', shard_generator=generator, tasks=tasks)
```

I find this confusing for a few reasons.  First, because the same function is used for two very different operations.  Second, because the default value for `reload` corresponds to the rare operation (creating a new dataset on disk) rather than the common one (loading an existing dataset).  Third, because it's not immediately obvious which of the two cases (or both) each argument is relevant to.

I suggest that the constructor be used only for loading an existing dataset:

```Python
dataset = DiskDataset('datadir')
```

We can then have a static method for creating a new dataset on disk:

```Python
dataset = DiskDataset.create_dataset(data_dir='datadir', shard_generator=generator, tasks=tasks)
```

**Comment by rbharath** [comment]
@peastman :+1: I like the proposed design. I've shot myself in the foot more than once misusing `reload`. I'd be very open to a PR implementing the change, or I could probably get to it in a couple weeks (after #293 and #313)

**Comment by rbharath** [comment]
Implemented in #330.
